### PR TITLE
Task av/j95 timestamp comment functionality

### PR DESCRIPTION
**Pull request by artem-totality**

### CODE REVIEW

**Review by JuliaDzyuba — DISMISSED** [review]

File: backend/src/api/records/records.api.ts [hunk 1]
```diff
@@ -20,6 +20,15 @@ const initRecordsApi = ({ apiRouter, recordService }: Args): Router => {
     }),
   );
 
+  recordRouter.get(
```
Comment (ghost) [inline backend/src/api/records/records.api.ts]: remove records api at all
you need to join record into episode

File: backend/src/data/repositories/record/record.repository.ts [hunk 2]
```diff
@@ -19,6 +19,10 @@ class Record {
   public create(payload: RecordCreatePayload): Promise<TRecord> {
     return this.#RecordModel.query().insert(payload);
   }
+
+  public getByEpisodeId(id: string): Promise<TRecord> {
+    return this.#RecordModel.query().findOne('episode_id', '=', id);
+  }
```
Comment (ghost) [inline backend/src/data/repositories/record/record.repository.ts]:
```suggestion
```

File: backend/src/services/record/record.service.ts [hunk 3]
```diff
@@ -19,6 +21,17 @@ class Record {
...
+
+  public async getByEpisodeId(id: string): Promise<TRecord> {
+    const record = await this.#recordRepository.getByEpisodeId(id);
+    if (!record) {
+      throw new HttpError({
+        status: HttpCode.NOT_FOUND,
+        message: ErrorMessage.RECORD_NOT_FOUND,
+      });
+    }
+    return record;
```
Comment (ghost) [inline backend/src/services/record/record.service.ts]:
```suggestion
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: frontend/src/components/episode/episode.tsx [hunk 4]
```diff
@@ -38,23 +56,28 @@ const Episode: React.FC = () => {
...
-              width="280"
-              height="280"
-              loading="lazy"
-              alt={episode.name}
-            />
-          </p>
-        </div>
+          {episode.record ? (
+            <Player src={episode.record.fileUrl} setRef={setPlayer} />
+          ) : null}
```
Comment (ghost) [inline frontend/src/components/episode/episode.tsx]:
```suggestion
          {episode.record && (
            <Player src={episode.record.fileUrl} setRef={setPlayer} />
          )}
```

File: frontend/src/components/episode/episode.tsx [hunk 5]
```diff
@@ -1,17 +1,26 @@
...
+  useState,
+} from 'hooks/hooks';
 import { episode as episodeActions } from 'store/actions';
 import { CreateCommentForm, CommentsList } from './components/components';
 import { Loader } from 'components/common/common';
 import { DataStatus } from 'common/enums/enums';
 import { CommentFormCreatePayload } from 'common/types/types';
 import { PageParams } from './common/types/types';
 import styles from './styles.module.scss';
+import { Player } from 'components/common/common';
```
Comment (ghost) [inline frontend/src/components/episode/episode.tsx]: we already have imports from components

File: frontend/src/components/episode/episode.tsx [hunk 6]
```diff
@@ -1,17 +1,26 @@
...
 
 const Episode: React.FC = () => {
   const dispatch = useDispatch();
   const { id } = useParams<PageParams>();
+  const [player, setPlayer] = useState<H5AudioPlayer | null>(null);
```
Comment (ghost) [inline frontend/src/components/episode/episode.tsx]:
```suggestion
  const [player, setPlayer] = useState<H5AudioPlayer | null>(null);
```
use useRef instead

File: frontend/src/components/episode/episode.tsx [hunk 7]
```diff
@@ -28,7 +37,16 @@ const Episode: React.FC = () => {
   }, []);
 
   const handleCreateComment = (payload: CommentFormCreatePayload): void => {
-    dispatch(episodeActions.createComment(payload));
+    const timestamp =
+      player && player.audio.current
```
Comment (ghost) [inline frontend/src/components/episode/episode.tsx]:
```suggestion
      Boolean(player?.audio.current)
```

File: frontend/src/components/episode/episode.tsx [hunk 8]
```diff
@@ -28,7 +37,16 @@ const Episode: React.FC = () => {
   }, []);
 
   const handleCreateComment = (payload: CommentFormCreatePayload): void => {
-    dispatch(episodeActions.createComment(payload));
+    const timestamp =
```
Comment (ghost) [inline frontend/src/components/episode/episode.tsx]: move this calculation to helper `frontend/src/components/episode/helpers/`

**Review by ghost — CHANGES_REQUESTED** [review]

File: frontend/src/components/episode/helpers/get-current-time.helper.ts [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import { PlayerRef } from 'components/common/player/player';
+
+const getCurrentTime = (playerRef: React.MutableRefObject<PlayerRef | null>): number => {
+  return playerRef.current
+    ? Math.round(playerRef.current?.getCurrentTime())
+    : 0;
```
Comment (ghost) [inline frontend/src/components/episode/helpers/get-current-time.helper.ts]:
```suggestion
const DEFAULT_TIME = 0;

const getCurrentTime = (playerRef: React.MutableRefObject<PlayerRef | null>): number => {
  return playerRef.current
    ? Math.round(playerRef.current?.getCurrentTime())
    : DEFAULT_TIME;
```

File: frontend/src/components/episode/helpers/get-current-time.helper.ts [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import { PlayerRef } from 'components/common/player/player';
```
Comment (ghost) [inline frontend/src/components/episode/helpers/get-current-time.helper.ts]: rename to `frontend/src/components/episode/helpers/get-current-time/get-current-time.helper.ts`

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by Denidzio — DISMISSED** [review]

File: frontend/src/components/episode/helpers/get-current-time/get-current-time.helper.ts [hunk 11]
```diff
@@ -1,9 +1,11 @@
 import { PlayerRef } from 'components/common/player/player';
 
 const getCurrentTime = (playerRef: React.MutableRefObject<PlayerRef | null>): number => {
+  const DEFAULT_TIME = 0;
```
Comment (Denidzio) [inline frontend/src/components/episode/helpers/get-current-time/get-current-time.helper.ts]: It would be better to move the constant out of the function

**Review by Denidzio — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**